File: php/testcode/benchmark_test_00704.php
<?php
require_once __DIR__ . '/shared.php';

function benchmarkTest00704(BenchmarkRequest $req): BenchmarkResponse {
    $data = $req->param('data');
    $key = random_bytes(SODIUM_CRYPTO_GENERICHASH_KEYBYTES);
    $hash = sodium_crypto_generichash($data, $key);
    return BenchmarkResponse::ok(bin2hex($hash));
}

File: rust/testcode/benchmark_test_00704.rs
fn ldap_search(base: &str, filter: &str) -> String {
    format!("LDAP search in {} with filter {}", base, filter)
}

pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let dept = req.param("dept");
    let valid_depts = ["Engineering", "Marketing", "Finance", "HR", "Operations"];
    if !valid_depts.contains(&dept.as_str()) {
        return super::shared::BenchmarkResponse::bad_request("unknown department");
    }
    let filter = format!("(ou={})", dept);
    let result = ldap_search("dc=example,dc=com", &filter);
    super::shared::BenchmarkResponse::ok(&result)
}
